kernel/rtlil.cc

```diff
@@ -597,6 +597,7 @@ void RTLIL::Design::remove(RTLIL::Module *module)
 	}
 
 	log_assert(modules_.at(module->name) == module);
+	log_assert(refcount_modules_ == 0);
```

Comment:
The whole reason `refcount_modules_` exists is for this assert. Why do we want to allow removing modules while there is an active ObjRange?